### PR TITLE
Update CI image & Envoy, move to clang-tidy-8

### PR DESCRIPTION
**Pull request by oschaaf**
- Update Envoy submodule to the latest
    - Use clang-8 for everything, including clang-tidy-8
    - Upgrade the CI image
    - Tidies up a few things, and drops the linker override hack (obsolete).

This ought to fail some CI tests, the next commit will fix that.
Fixes https://github.com/envoyproxy/nighthawk/issues/62

Signed-off-by: Otto van der Schaaf <oschaaf@we-amp.com>

### CODE REVIEW

File: test/statistic_test.cc [hunk 1]
```diff
@@ -210,6 +210,7 @@ TEST(StatisticTest, SimpleStatisticProtoOutputLargeValues) {
   const nighthawk::client::Statistic proto = a.toProto();
 
   EXPECT_EQ(proto.count(), 2);
+  // NOLINTNEXTLINE
```
Comment (htuch) [inline test/statistic_test.cc]: How come these?

File: test/client_test.cc [hunk 2]
```diff
@@ -103,8 +103,8 @@ class ClientTest : public Envoy::BaseIntegrationTest,
 
   int port_;
   pid_t pid_;
-  int fd_port_[2];
-  int fd_confirm_[2];
+  int fd_port_[2];    // NOLINT
+  int fd_confirm_[2]; // NOLINT
```
Comment (htuch) [inline test/client_test.cc]: ?

**Comment by oschaaf** [comment]
@htuch I added a couple of changes to avoid the NOLINTs in https://github.com/envoyproxy/nighthawk/pull/64/commits/7c5844f6eeb88e329fa493c664bf86afd117f7ff

Ready for another pass.

File: test/client_test.cc [hunk 3]
```diff
@@ -45,8 +45,8 @@ class ClientTest : public Envoy::BaseIntegrationTest,
     // runtimeloaders as both NH and the integration server want to own that and we can have only
     // one. The plan is to move to python for this type of testing, so hopefully we can deprecate
     // this test and it's peculiar setup with fork/pipe soon.
-    RELEASE_ASSERT(pipe(fd_port_) == 0, "Failed to open pipe");
-    RELEASE_ASSERT(pipe(fd_confirm_) == 0, "Failed to open pipe");
+    RELEASE_ASSERT(pipe(&fd_port_[0]) == 0, "Failed to open pipe");
```
Comment (htuch) [inline test/client_test.cc]: `fd_port_.data()` (and below)

File: test/client_test.cc [hunk 4]
```diff
@@ -103,8 +103,8 @@ class ClientTest : public Envoy::BaseIntegrationTest,
 
   int port_;
   pid_t pid_;
-  int fd_port_[2];
-  int fd_confirm_[2];
+  std::vector<int> fd_port_ = {0, 0};
```
Comment (htuch) [inline test/client_test.cc]: You can't do an explicit initialization assignment to an array?

**Comment by oschaaf** [comment]
@htuch updated

**Pull request merged**